build/patches/Timezone-Override.patch

```diff
@@ -0,0 +1,1379 @@
+new file mode 100755
+--- /dev/null
++++ b/components/browser_ui/site_settings/android/java/src/org/chromium/components/browser_ui/site_settings/TimezoneOverrideSiteSettingsPreference.java
+@@ -0,0 +1,193 @@
++// Copyright 2018 The Chromium Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style license that can be
++// found in the LICENSE file.
++
++package org.chromium.components.browser_ui.site_settings;
++
++import org.chromium.base.Log;
++import android.content.Context;
++import android.content.Intent;
++import android.content.DialogInterface;
++import android.view.View;
++import android.widget.RadioGroup;
++import android.widget.Button;
++import android.widget.TextView;
++import android.util.AttributeSet;
++import android.widget.ArrayAdapter;
++import android.widget.ListView;
++import android.view.LayoutInflater;
++import android.widget.AdapterView;
++import android.graphics.Color;
++
++import androidx.preference.Preference;
++import androidx.preference.PreferenceViewHolder;
++import androidx.appcompat.app.AlertDialog;
++
++import org.chromium.components.content_settings.ContentSettingValues;
++import org.chromium.components.embedder_support.browser_context.BrowserContextHandle;
++import org.chromium.components.browser_ui.site_settings.WebsitePreferenceBridge;
++import org.chromium.components.browser_ui.widget.RadioButtonWithDescription;
++import org.chromium.components.browser_ui.widget.RadioButtonWithEditText;
++import org.chromium.components.browser_ui.widget.TintedDrawable;
++
++import java.util.ArrayList;
++import java.util.Arrays;
++import java.util.TimeZone;
++
++/**
++ * TimezoneOverride Preference for SiteSettings.
++ */
++public class TimezoneOverrideSiteSettingsPreference
++        extends Preference implements RadioGroup.OnCheckedChangeListener,
++                                      RadioButtonWithEditText.OnTextChangeListener {
++    private @ContentSettingValues int mSetting = ContentSettingValues.DEFAULT;
++    private RadioButtonWithDescription mAllowed;
++    private RadioButtonWithEditText mAsk;
++    private RadioButtonWithDescription mBlocked;
++    private RadioGroup mRadioGroup;
++    private TextView mSelectButton;
++
++    private String currentSelected;
++
++    private BrowserContextHandle mBrowserContextHandle;
++
++    // reference for translation
++    // 4746227879892920228 "Allow sites to use System Timezone"
++    // 8984679974863410689 "SELECT TIMEZONE FROM LIST"
++    // 3394263620263751240 "bla bla bla"
++    // 5477585238714637279 "Select Timezone"
++    // 6450629284808068080 "Use custom timezone"
++    // 1774419748905152303 "Random Value"
++    // 6315329332061924002 "Default UTC"
++    // 7790128239055164249 "Timezone override"
++    // 2520215895798545849 "Use System Timezone"
++    // 7337126996643069922 "Use random timezone for each page"
++    // remember, hash change if you change default text (not key!)
```

Comment:
Remind me, how do you calculate the hashes? I have been trying to script this.

Reply:
sure, you need

```
cd <directory of grd>
~/chromium/src/tools/grit/grit.py -v -i <file with extension .grd> build
```

so you can see in stdout the new keys